include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm_banded.hpp

```diff
@@ -85,18 +85,24 @@ class pairwise_alignment_algorithm_banded :
 
         for (auto && [sequence_pair, idx] : indexed_sequence_pairs)
         {
-            size_t const sequence1_size = std::ranges::distance(get<0>(sequence_pair));
+            size_t sequence1_size = std::ranges::distance(get<0>(sequence_pair));
             size_t const sequence2_size = std::ranges::distance(get<1>(sequence_pair));
 
+            auto && [alignment_matrix, index_matrix] = this->acquire_matrices(sequence1_size,
+                                                                              sequence2_size,
+                                                                              this->lowest_viable_score());
+
             // Initialise the cell updater with the dimensions of the regular matrix.
             this->compare_and_set_optimum.set_target_indices(row_index_type{sequence2_size},
                                                              column_index_type{sequence1_size});
 
-            auto && [alignment_matrix, index_matrix] = this->acquire_matrices(sequence1_size,
-                                                                              sequence2_size,
-                                                                              this->lowest_viable_score());
+            // Shrink the first sequence if the band ends before its actual end.
+            sequence1_size = std::min(sequence1_size, this->upper_diagonal + sequence2_size);
 
-            compute_matrix(get<0>(sequence_pair), get<1>(sequence_pair), alignment_matrix, index_matrix);
+            compute_matrix(get<0>(sequence_pair) | views::take(sequence1_size),
+                           get<1>(sequence_pair),
```

Comment:
So this can only happen with the upper_diagonal and sequence1 and not with the lower_diagonal and sequence2?

Reply:
yes. The point is that we have a column based approach such that the last column might be different in the band. The last row might be different as well, but then the column is limited by the band. This is already part of the algorithm but happens inside of the function compute_matrix